providers/implementations/keymgmt/ml_kem_kmgmt.c

```diff
@@ -0,0 +1,631 @@
+/*
+ * Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/core_dispatch.h>
+#include <openssl/core_names.h>
+#include <openssl/params.h>
+#include <openssl/err.h>
+#include <openssl/proverr.h>
+#include <openssl/rand.h>
+#include <openssl/self_test.h>
+#include "internal/param_build_set.h"
+#include <openssl/param_build.h>
+#include "prov/ml_kem.h"
+#include "prov/implementations.h"
+#include "prov/providercommon.h"
+#include "prov/provider_ctx.h"
+#include "prov/securitycheck.h"
+#include <assert.h>
+
+static OSSL_FUNC_keymgmt_free_fn ml_kem_free;
+static OSSL_FUNC_keymgmt_gen_fn ml_kem_gen;
+static OSSL_FUNC_keymgmt_gen_cleanup_fn ml_kem_gen_cleanup;
+static OSSL_FUNC_keymgmt_gen_set_params_fn ml_kem_gen_set_params;
+static OSSL_FUNC_keymgmt_gen_settable_params_fn ml_kem_gen_settable_params;
+static OSSL_FUNC_keymgmt_get_params_fn ml_kem_get_params;
+static OSSL_FUNC_keymgmt_gettable_params_fn ml_kem_gettable_params;
+static OSSL_FUNC_keymgmt_set_params_fn ml_kem_set_params;
+static OSSL_FUNC_keymgmt_settable_params_fn ml_kem_settable_params;
+static OSSL_FUNC_keymgmt_has_fn ml_kem_has;
+static OSSL_FUNC_keymgmt_match_fn ml_kem_match;
+static OSSL_FUNC_keymgmt_import_fn ml_kem_import;
+static OSSL_FUNC_keymgmt_export_fn ml_kem_export;
+static OSSL_FUNC_keymgmt_import_types_fn ml_kem_imexport_types;
+static OSSL_FUNC_keymgmt_export_types_fn ml_kem_imexport_types;
+static OSSL_FUNC_keymgmt_dup_fn ml_kem_dup;
+
+typedef const ossl_ml_kem_vinfo *vinfo_t;
```

Comment:
Look like this should be in a header that gets included.

Reply:
I'll leave this open for discussion in #26172